x-pack/plugins/security_solution/public/common/components/navigation/use_get_url_search.tsx

```diff
@@ -6,17 +6,18 @@
  */
 
 import { useCallback, useMemo } from 'react';
+import type { SecurityPageName } from '../../../app/types';
 
 import { useGlobalQueryString } from '../../utils/global_query_string';
 
 import { getSearch } from './helpers';
 import type { SearchNavTab } from './types';
 
-export const useGetUrlSearch = (tab?: SearchNavTab) => {
+export const useGetUrlSearch = (pageName: SecurityPageName, tab?: SearchNavTab) => {
   const globalQueryString = useGlobalQueryString();
   const urlSearch = useMemo(
-    () => (tab ? getSearch(tab, globalQueryString) : ''),
-    [tab, globalQueryString]
+    () => (tab ? getSearch(pageName, globalQueryString) : ''),
```

Comment:
I don't understand this `tab ? ...`. Is it needed?

Reply:
the `getSearch` function is only one line now, can we remove it and check the `needsUrlState` here directly?

Reply:
> I don't understand this `tab ? ...`. Is it needed?

Nice catch. After my changes, we can delete `tab`.

Reply:
> the `getSearch` function is only one line now, can we remove it and check the `needsUrlState` here directly?

I gree with you, but for now I am keeping the function because it also adds `?` to the beginning of the string.